File: 0237_delete_node_in_linked_list/delete_node.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 * Time: O(1)
 * Space: O(1)
 */
func deleteNode(node *ListNode) {
	next := node.Next
	node.Val = next.Val
	node.Next = next.Next
}

func main() {

}

File: 0237_delete_node_in_linked_list/delete_node.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


def delete_node(node: ListNode) -> None:
    """
    Time: O(1)
    Space: O(1)
    """
    _next = node.next
    node.val = _next.val
    node.next = _next.next
